### PR TITLE
generate.sh -> Makefile

### PR DESCRIPTION
**Pull request by liyishuai**

### CODE REVIEW

**Comment by Zimmi48** [comment]
If you do this change, you need to fix the README. Can you briefly summarize the motivation?

**Comment by liyishuai** [comment]
I see. I should change the configuration to assume `templates` in a subdirectory.

**Comment by palmskog** [comment]
My general argument is basically that I don't think a Makefile is flexible enough to accommodate the many workflows which one could incorporate the templates into. In particular, I think we need some kind of solution that works both when `templates` is in a subdirectory (of the current Coq project) and when it lives somewhere completely different.

The parametrization (pattern-matching, really) of Makefiles (`%.mustache` and so on) is also lacking. The current `generate.sh` script is not great at all, but there should at least be some improvement on the above issues for a replacement.

**Comment by liyishuai** [comment]
Changes:
- Moved example to subfolder
- Specific rules for `.opam` files
- Replaced target `index.md` with `index.html`

File: example/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.travis.yml
+README.md
+coq-chapar-stores.opam
+coq-chapar.opam
+default.nix
+index.html
+index.md
```
Comment (liyishuai) [inline example/.gitignore]: Maybe we should neither ignore `index.md` or remove it in `make clean`, as this file is subject to changes that repos should keep track of.

**Comment by Zimmi48** [comment]
If I understood correctly, this PR has been made obsolete by #26 and all that remains to do is to address #20 and #24.

**Pull request closed without merging**